### PR TITLE
New board definition: Piconomix PX-HER0 board (STM32L072RB)

### PR DESCRIPTION
**Pull request by pieterconradie**
Hi there!

This is my first time working with stm32duino and adding a new variant, so all pointers / tips / hints will be much appreciated!

Thanks in advance,
Pieter
https://piconomix.com

### CODE REVIEW

**Comment by fpistm** [comment]
Hi @pieterconradie 
thanks for this PR.
I will made a review.

File: boards.txt [hunk 1]
```diff
@@ -1453,6 +1453,89 @@ Midatronics.menu.upload_method.dfuMethod.upload.protocol=2
 Midatronics.menu.upload_method.dfuMethod.upload.options=-g
 Midatronics.menu.upload_method.dfuMethod.upload.tool=stm32CubeProg
 
+################################################################################
+# Piconomix boards
```
Comment (fpistm) [inline boards.txt]: Please add this board in the GenL0 submenu.

File: boards.txt [hunk 2]
```diff
@@ -1453,6 +1453,89 @@ Midatronics.menu.upload_method.dfuMethod.upload.protocol=2
...
+
+# PX-HER0
+Piconomix.menu.pnum.PX-HER0=PX-HER0
+Piconomix.menu.pnum.PX-HER0.node=NODE_L072RB
+Piconomix.menu.pnum.PX-HER0.build.board=PX-HER0
+Piconomix.menu.pnum.PX-HER0.build.mcu=cortex-m0plus
+Piconomix.menu.pnum.PX-HER0.build.cmsis_lib_gcc=arm_cortexM0l_math
+Piconomix.menu.pnum.PX-HER0.build.series=STM32L0xx
+Piconomix.menu.pnum.PX-HER0.build.product_line=STM32L072xx
+Piconomix.menu.pnum.PX-HER0.build.variant=PX-HER0
```
Comment (fpistm) [inline boards.txt]: The build is not passed because the directory is `PX_HER0` not `PX-HER0`

File: boards.txt [hunk 3]
```diff
@@ -1453,6 +1453,89 @@ Midatronics.menu.upload_method.dfuMethod.upload.protocol=2
...
+Piconomix.name=Piconomix boards
+
+Piconomix.build.vid=0x0483
+Piconomix.build.core=arduino
+Piconomix.build.board=Piconomix
+Piconomix.build.extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial} -D__CORTEX_SC=0
+
+# PX-HER0
+Piconomix.menu.pnum.PX-HER0=PX-HER0
+Piconomix.menu.pnum.PX-HER0.node=NODE_L072RB
```
Comment (fpistm) [inline boards.txt]: `node` is not needed. It i used for `massStorageCopy` upload method.

**Comment by fpistm** [comment]
@pieterconradie 
Thanks for the update.
Anyway, I would prefer `PX_HER0` instead of `PX-HER0`.
I don't know how the "-" could be interpreted if it is used in a command line. So please use "_".

**Comment by pieterconradie** [comment]
Hi @fpistm Thanks for the guidance :) The "astyle.py" script did not work for me so I ran Astyle manually to fix the files. Hope all is OK now. Thanks! Pieter

**Comment by pieterconradie** [comment]
Hi @fpistm I got the astyle.py script working for me and the pull request is now passing the CI checks :) Best regards, Pieter

File: variants/PX_HER0/variant.h [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+#define PB14                38     /* PX_SPI2_MISO */
+#define PA4                 39     /* PX_DAC1 */
+#define PB0                 40     /* PX_ADC0 */
+#define PC5                 41     /* PX_ADC1 */
+#define PC4                 42     /* PX_ADC2 */
+#define PA7                 43     /* PX_ADC3 */
+#define PB1                 44     /* PX_ADC_VBAT */
+
+// Mapping of net name to digital pin number
+#define PX_GPIO0            0
```
Comment (fpistm) [inline variants/PX_HER0/variant.h]: I advise to use `PYn` notation for better readability.
But it's up to you :wink:
```suggestion
#define PX_GPIO0            PA6
```

File: variants/PX_HER0/variant.h [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+#define PIN_SERIAL2_TX          PA2
+#define PIN_SERIAL2_RX          PA3
+#define PIN_SERIAL4_TX          PC10
+#define PIN_SERIAL4_RX          PC11
+#define PIN_SERIALLP1_TX        PA2
+#define PIN_SERIALLP1_RX        PA3
+
+// SD card slot Definitions
+// SD detect signal can be defined if required
+#define SD_DETECT_PIN           PB7
```
Comment (fpistm) [inline variants/PX_HER0/variant.h]: This define is required for the STM32SD library which use the SDIO or SDMMC interface.
As this MCU does not have this and you use an SPI card slot you could remove it:
```suggestion
```
Same for all SD commented related stuff

**Comment by pieterconradie** [comment]
Hi @fpistm Thanks for the advice :) I have made the changes. If you are interested the board design files (including schematic) are [HERE](https://github.com/piconomix/piconomix-fwlib/tree/master/boards/arm/stm32/piconomix_hero_board/docs). My test sketches are [HERE](https://github.com/piconomix/piconomix-fwlib/tree/master/boards/arm/stm32/piconomix_hero_board/arduino/examples)

**Comment by fpistm** [comment]
I will merge it and add a reference in the README.md. Could you give me a link I can add to reference the board, please?

**Comment by pieterconradie** [comment]
Hi @fpistm Thank you kindly for your help and merging the code :) [HERE](https://piconomix.com/fwlib/_p_i_c_o_n_o_m_i_x__s_t_m32__h_e_r_o__b_o_a_r_d.html) is a link to the board.

**Comment by fpistm** [comment]
@pieterconradie 
Thanks, The README.md has been updated.

**Pull request merged**